### PR TITLE
ASoC: intel:sdw_rt711_i2s_rt1308: add quirk for rt711

### PR DESCRIPTION
**Pull request by RanderWang**
Set rt711 device property on machine driver according to DMI.
On TGL, JD1 is used for headphone.

pick https://github.com/thesofproject/linux/pull/1427 to sdw_rt711_i2s_rt1308 for tigerlake

it fixes https://github.com/thesofproject/linux/issues/1439

### CODE REVIEW

File: sound/soc/intel/boards/sdw_rt711_i2s_rt1308.c [hunk 1]
```diff
@@ -192,6 +205,49 @@ static struct snd_soc_ops rt1308_ops = {
...
+{
+	sof_rt711_rt1308_quirk = (unsigned long)id->driver_data;
+	return 1;
+}
+
+static const struct dmi_system_id sof_sdw_rt711_rt1308_quirk_table[] = {
+	{
+		.callback = sof_rt711_rt1308_quirk_cb,
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "IntelCorporation"),
```
Comment (plbossart) [inline sound/soc/intel/boards/sdw_rt711_i2s_rt1308.c:218]: spaces?
and board name?

existing examples are:

.matches = {
			DMI_MATCH(DMI_SYS_VENDOR, "Intel Corporation"),
			DMI_MATCH(DMI_PRODUCT_NAME, "WhiskeyLake Client"),
		},

Comment (RanderWang) [reply sound/soc/intel/boards/sdw_rt711_i2s_rt1308.c:218]: Thanks, I checked there is no space.   I will add product name

**Comment by RanderWang** [comment]
Thanks!  I update product name.
Tested on TGL RVP

File: sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c [hunk 2]
```diff
@@ -29,8 +31,6 @@
 
 #define MAX_NO_PROPS 2
 
-extern struct bus_type sdw_bus_type;
-
```
Comment (plbossart) [inline sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c:33]: can you move this to a different patch, it makes my life easier when I squash patches...

**Comment by RanderWang** [comment]
Thanks!  update my PR to two patches.

File: sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c (hunk 2 above)
Comment (RanderWang) [reply sound/soc/intel/boards/sdw_rt711_rt1308_rt715.c:33]: yes, done!
